pkg/concurrency/runner.go

```diff
@@ -0,0 +1,89 @@
+package concurrency
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// Runner is a function that runs a task.
+type Runner func(context.Context) error
+
+// RunnerManager is a manager for runners. It runs all runners in parallel and
+// waits for all runners to finish. If any runner returns, the RunnerManager
+// will stop all other runners and return any error.
+type RunnerManager struct {
+	errs    []string
+	lock    sync.Mutex
+	wg      sync.WaitGroup
+	runners []Runner
+	running chan struct{}
```

Comment:
isn't an atomic bool enough for this scenario?

fyi:https://pkg.go.dev/sync/atomic

Reply:
I agree. A Compare-and-swap may be simpler as semaphore